src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueueClient.cs

```diff
@@ -41,6 +49,7 @@ public class AzureServiceBusQueueClient : BaseQueueClient
                 });
 
             _messageReceiver = _queue.MessageReceiver;
+            _messageReceiver.PrefetchCount = prefetchCount;
```

Comment:
This line will change the prefetch count of existing `_messageReceiver`s: if `GetQueueReference` is called multiple times with different values for the prefetch count, the last call will win. Is that what we expect should happen? That might be surprising for users of the library.

Instead, should this value form part of the key that we use to cache instances of `_messageReceiver`s? Then, if the same combination of prefetch count and connection string is used, we reuse a cached `_messageReceiver`, otherwise a new `_messageReceiver` is created.

Reply:
Since the prefetch count is a readonly field of the AzureServiceBusQueueClientFactory it won't change between calls of `GetQueueReference`. It is set (or not set) once in the constructor and cannot be changed after that.